### ISSUE TITLE
Help Request: Unable to Receive WhatsApp Messages Using Code

### ISSUE THREAD

**Issue by jcabourne** [filipporomani/whatsapp#21]
Hi ,

I hope this message finds you well. I am writing to seek assistance with an issue I am encountering while trying to send WhatsApp messages using a custom Python script. I have provided the relevant code snippet below:

```python
from whatsapp import WhatsApp, Message

if __name__ == "__main__":
    whatsapp_token = 'EAAVXKQbbM7**uZBj'
    messenger = WhatsApp(token=whatsapp_token, phone_number_id='1830**0')
    msg = Message(instance=messenger, content="Hello World!", to="+44***91")
    response = msg.send()
    print(response)
```

The issue I am facing is that the script runs successfully without any errors, and the logs indicate that the message has been sent:

```
2024-01-26 15:32:07,236 - root - INFO - Sending message to +44**91
2024-01-26 15:32:08,044 - root - INFO - Message sent to +44***91
```

However, despite the successful message send logs, I am not receiving any messages on WhatsApp. It's worth noting that I do receive messages when testing using the pre-written functions in the Facebook Developer Portal.

I would greatly appreciate any guidance or suggestions you can provide to help me resolve this issue. If there are additional logs or information you need from my end, please let me know.

Thank you in advance for your assistance.

**Comment by filipporomani** [comment]
Hello! I'm sorry for the inconvenience and thanks for reporting that!

Can you please try to send a message to the number you are using in the python code with your own WhatsApp number?

After sending a message to that number, do you receive messages?

**Comment by jcabourne** [comment]
Hi thanks for getting back to me. I am currently using my own number for testing and I am not receiving any messages after being confirmed in the log.

**Comment by filipporomani** [comment]
So you are trying to send an SMS to your own number? I'm not sure if that's possible using the API! 

But if it works from the website, then it should also work in the python code, I get it. 

The problem is:
From the website, you send a `template`, which can be sent to any number in the world without any restriction. 
In your code, you are sending a plain text message, that cannot be sent to everybody. To send a custom message (the one you are trying to send unsuccessfully) to an user the user must send a message to the automation first, so the conversation must already be initiated. 

The API returns no error in that case tho, so you get the sent message log. 

Idk if this helps or not, let me know or try to describe the situation better!

**Comment by jcabourne** [comment]
Ok that's for letting me know. Is it possible to do temples with your python wrapper to initiate new conversations? If I wanted to send a message to a group do you know how that would work?

**Comment by filipporomani** [comment]
Yes of course! You can use the [send_template() method](https://github.com/filipporomani/whatsapp/wiki/Sending-templates) to do so. To create templates you have to use the Facebook Developer portal, and they need to get approved. 

Also, just to remind: you'll be able to send custom plain text messages only after the receiver **replies to the template**, and that will last for 24 hours. After that, user will need to send a message to your application again.

**Comment by filipporomani** [comment]
To be honest i never tried sending messages to a group, I suggest you to setup the incoming message event listener and to send a message in that group after you add the app to the chat.
You can retreive the sender ID and maybe try to send the message to that ID.
Don't forget to add `rec_type="group"` when initializing the `Message()` object if you want to send the message to a group.

**Comment by filipporomani** [comment]
Closing this as completed, feel free to reopen in case you need further assistance!
